pkg/worker/worker.go

```diff
@@ -249,17 +250,22 @@ func (w *Worker) setupTests(ex executor.Executor, workDir, repoDir string) (bool
 		klog.V(3).Infof("PR %s and main branch %s", w.cimsg.Target, w.cimsg.MainBranch)
 		chkout = fmt.Sprintf("pr%s", w.cimsg.Target)
 		pulltgt := fmt.Sprintf("pull/%s/head:%s", w.cimsg.Target, chkout)
-		status, err = w.runCommand(status, ex, repoDir, []string{"git", "fetch", "-v", "origin", pulltgt})
+		status1, err := w.runCommand(status, ex, repoDir, []string{"git", "fetch", "-v", "origin", pulltgt})
 		if err != nil {
 			return false, fmt.Errorf("failed to fetch pr no %s, are you sure it exists in repo %s %w", w.cimsg.Target, w.cimsg.RepoURL, err)
 		}
-		status, err = w.runCommand(status, ex, repoDir, []string{"git", "checkout", w.cimsg.MainBranch})
+		if !status1 {
+			fmt.Printf("couldn't find remote ref for pr no %s, running tests on main branch", w.cimsg.Target)
+		}
+		status, err = w.runCommand(true, ex, repoDir, []string{"git", "checkout", w.cimsg.MainBranch})
 		if err != nil {
 			return false, fmt.Errorf("failed to switch to main branch %w", err)
 		}
-		status, err = w.runCommand(status, ex, repoDir, []string{"git", "merge", chkout, "--no-edit"})
-		if err != nil {
-			return false, fmt.Errorf("failed to fast forward merge %w", err)
+		if status1 {
```

Comment:
can we do this? this way we will not need to add new variables to the code.
```
status, err := w.runCommand(status, ex, repoDir, []string{"git", "fetch", "-v", "origin", pulltgt})
		if err != nil {
			return false, fmt.Errorf("failed to fetch pr no %s, are you sure it exists in repo %s %w", w.cimsg.Target, w.cimsg.RepoURL, err)
		}
		if !status {
			fmt.Printf("couldn't find remote ref for pr no %s, running tests on main branch", w.cimsg.Target)
		}
		_, err = w.runCommand(true, ex, repoDir, []string{"git", "checkout", w.cimsg.MainBranch})
		if err != nil {
			return false, fmt.Errorf("failed to switch to main branch %w", err)
		}
		if status {
			status, err = w.runCommand(status, ex, repoDir, []string{"git", "merge", chkout, "--no-edit"})
			if err != nil {
				return false, fmt.Errorf("failed to fast forward merge %w", err)
			}
		}
```

Reply:
> can we do this? this way we will not need to add new variables to the code.

no, I don't think so because `status, err = w.runCommand(status, ex, repoDir, []string{"git", "merge", chkout, "--no-edit"` will take only the status of `git checkout mainbranch` . Imagine the case when git fetch pass but git checkout status gives us `false` . In that scenario `git merge` should pass `false` as an argument.

Reply:
ooh yes understood. 
Thanks!